File: src/main/java/quizapi/model/Record.java
package quizapi.model;

import javax.persistence.*;

@Entity
@Table(name = "records")
public class Record {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", unique = true, nullable = false)
    private Long id;

    @Column(name = "score", nullable = false)
    private Integer score;

    @Column(name = "question_count", nullable = false)
    private Integer questionCount;

    @ManyToOne(targetEntity = Category.class)
    private Category category;

    @ManyToOne(targetEntity = User.class)
    private User user;

    public Record() {
    }

    public Record(Integer score,Integer questionCount, Category category, User user) {
        this.score = score;
        this.questionCount = questionCount;
        this.category = category;
        this.user = user;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Integer getScore() {
        return score;
    }

    public void setScore(Integer score) {
        this.score = score;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}

File: src/main/kotlin/com/shkila/quizapi/model/Record.kt
package com.shkila.quizapi.model

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.ManyToOne
import javax.persistence.Table

@Entity
@Table(name = "records")
data class Record(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "id", unique = true, nullable = false)
        val id: Long? = null,

        @Column(name = "score", nullable = false)
        var score: Int? = null,

        @Column(name = "question_count", nullable = false)
        var questionCount: Int? = null,

        @ManyToOne(targetEntity = Category::class)
        var category: Category? = null,

        @ManyToOne(targetEntity = User::class)
        var user: User? = null
)